### PR TITLE
feat(cat-voices): My account, modals

### PR DESCRIPTION
**Pull request by digitalheartxs**
# Description

Modals in My account

## Related Issue(s)

Closes https://github.com/input-output-hk/catalyst-voices/issues/822

## Screenshots

https://github.com/user-attachments/assets/3114feb1-bdc8-4c42-8f97-85f73916f7e0

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/pages/account/keychain_deleted_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                  const SizedBox(height: 32),
+                  Text(
+                    '''
+We reverted this device to Catalyst first use.''',
+                    style: Theme.of(context).textTheme.bodyMedium,
+                    textAlign: TextAlign.center,
+                  ),
+                  const SizedBox(height: 24),
+                  VoicesFilledButton(
+                    //backgroundColor: Theme.of(context).colors.iconsError,
```
Comment (damian-molinski) [inline catalyst_voices/lib/pages/account/keychain_deleted_dialog.dart:43]: delete

File: catalyst_voices/lib/pages/account/keychain_deleted_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    onTap: () => Navigator.of(context).pop(),
+                    child: const Text('Close'),
+                  ),
+                ],
+              ),
+            ),
+          ),
+          const DialogCloseButton(),
```
Comment (damian-molinski) [inline catalyst_voices/lib/pages/account/keychain_deleted_dialog.dart:51]: This pattern of **Stack** and **DialogCloseButton** is at least in 3 places already. How about extracting it into widget?

File: catalyst_voices/lib/pages/account/delete_keychain_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                        ),
+                      ),
+                    ],
+                  ),
+                  const SizedBox(height: 24),
+                  Wrap(
+                    children: [
+                      VoicesFilledButton(
+                        backgroundColor: Theme.of(context).colors.iconsError,
+                        onTap: () async {
```
Comment (damian-molinski) [inline catalyst_voices/lib/pages/account/delete_keychain_dialog.dart:120]: extract this onTap function into private one in **_DeleteKeychainDialogState** 

smth like this
```dart
VoicesFilledButton(
onTap: _verygoodname
);

Future<void> _verygoodname(){
                          if (_textEditingController.text ==
                              'Remove Keychain') {
                            // TODO(Jakub): remove keychain
                            Navigator.of(context).pop();
                            await VoicesDialog.show<void>(
                              context: context,
                              builder: (context) {
                                return const KeychainDeletedDialog();
                              },
                            );
                          } else {
                            setState(() {
                              _errorText = 'Text incorrect';
                            });
                          }
}
```

File: catalyst_voices/lib/pages/account/delete_keychain_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                          if (_textEditingController.text ==
+                              'Remove Keychain') {
```
Comment (damian-molinski) [inline catalyst_voices/lib/pages/account/delete_keychain_dialog.dart:122]: "Remove Keychain" probably should be private constant at the top of file. Seems important.

File: catalyst_voices/lib/pages/account/delete_keychain_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                            // TODO(Jakub): remove keychain
+                            Navigator.of(context).pop();
+                            await VoicesDialog.show<void>(
+                              context: context,
+                              builder: (context) {
+                                return const KeychainDeletedDialog();
+                              },
+                            );
```
Comment (damian-molinski) [inline catalyst_voices/lib/pages/account/delete_keychain_dialog.dart:130]: Does it make sense for this dialog to open another one? Maybe it should be better for this to, lets say, return "true" with pop `Navigator.of(context).pop(true)` 

```dart
final confirmed = await DeleteKeychainConfirmationDialog.show(context);

if(confirmed && mounted){
     KeychainDeletedDialog.show(context);
}

```

**Review by damian-molinski — COMMENTED** [review]
Extract all strings into l10n. It does not seems it will be mocked

File: catalyst_voices/lib/pages/account/delete_keychain_dialog.dart (hunk 5 above)
Comment (digitalheartxs) [reply catalyst_voices/lib/pages/account/delete_keychain_dialog.dart]: I find passing value like that messy, however, I agree that dialog shouldn't open another one so I've introduced `onDeleteKeychain` callback

File: catalyst_voices/lib/pages/account/account_page.dart [hunk 6]
```diff
@@ -35,7 +38,25 @@ final class AccountPage extends StatelessWidget {
...
-                    onRemoveKeychain: () => debugPrint('Keychain removed'),
+                    onRemoveKeychain: () async {
+                      await VoicesDialog.show<void>(
+                        context: context,
+                        builder: (context) {
+                          return DeleteKeychainDialog(
+                            onRemoveKeychainConfirmed: () async {
+                              // TODO(Jakub): remove keychain
+                              Navigator.of(context).pop();
+                              await VoicesDialog.show<void>(
```
Comment (damian-molinski) [inline catalyst_voices/lib/pages/account/account_page.dart:49]: Problem here is that you're using `context` from `DeleteKeychainDialog` dialog, which is going to be disposed soon. Probably its not end of the world but this is not good practice and nesting is bad :D

File: catalyst_voices/lib/pages/account/delete_keychain_dialog.dart (hunk 5 above)
Comment (dtscalac) [reply catalyst_voices/lib/pages/account/delete_keychain_dialog.dart]: Returning value from dialog, it's quite common, i.e. `showDatePicker()` from flutter has return type of `Future<DateTime?>`.

But lets do what Damian suggested, the messy part is that `showDialog` method has generic return type and will accept any type. To make it not messy we need a wrapper that will show the dialog for us and return expected type. You can see that on `DeleteKeychainConfirmationDialog.show(context);`

You would define it as:

```dart
class DeleteKeychainConfirmationDialog {
  static Future<bool?> show(BuildContext context) {
    // show dialog and cast the type
  }
}
```

Comment (damian-molinski) [reply catalyst_voices/lib/pages/account/delete_keychain_dialog.dart]: Its also every easy to do smth like this

```dart
class DeleteKeychainConfirmationDialog {
  static Future<bool> show(BuildContext context) async {
    final result =await  VoicesDialog.show<bool>(
      context: context,
      builder: (context) => ...,
    );
    
    return result ?? false;
  }
}
```

Comment (digitalheartxs) [reply catalyst_voices/lib/pages/account/delete_keychain_dialog.dart]: It's not fully clear for me. Would you like every dialog to have its own static `show` function?
It's simple `bool` type so what we need to map it to? Do you mean that this function will have named bool and that's why it's gonna be more semantic?

**Review by damian-molinski — APPROVED** [review]
lgtm

**Pull request merged**